editor/scene_create_dialog.cpp

```diff
@@ -0,0 +1,255 @@
+		update_error(file_error_label, MSG_ERROR, TTR("File already exists."));
+		is_valid = false;
+	}
+
+	node_type_custom->set_icon(get_theme_icon(root_custom_type->get_text(), SNAME("EditorIcons")));
+	// int idx = EditorNode::get_singleton()->new_scene();
+	// EditorNode::get_singleton()->get_editor_data().set_scene_path(idx, scene_name);
+
+	// update_error(file_error_label, true, TTR("File name is valid."));
+	// node_error_label->hide();
+	// update_error(node_error_label, true, TTR("File name is valid."));
+}
+
+void SceneCreateDialog::update_error(Label *p_label, MsgType p_type, const String &p_msg) {
+	p_label->set_text(String::utf8("•  ") + p_msg);
+	switch (p_type) {
+		case MSG_OK:
+			p_label->add_theme_color_override("font_color", get_theme_color(SNAME("success_color"), SNAME("Editor")));
+		case MSG_WARNING:
+			p_label->add_theme_color_override("font_color", get_theme_color(SNAME("warning_color"), SNAME("Editor")));
+		case MSG_ERROR:
+			p_label->add_theme_color_override("font_color", get_theme_color(SNAME("error_color"), SNAME("Editor")));
+	}
+}
+
+void SceneCreateDialog::_bind_methods() {
+	ADD_SIGNAL(MethodInfo("scene_created", PropertyInfo(Variant::OBJECT, "scene", PROPERTY_HINT_RESOURCE_TYPE, "PackedScene")));
+}
+
+SceneCreateDialog::SceneCreateDialog() {
+	select_node = memnew(CreateDialog);
+	add_child(select_node);
+	select_node->connect("create", callable_mp(this, &SceneCreateDialog::type_picked));
+	select_node->set_base_type("Node");
+	select_node->set_title(TTR("Pick Root Node Type"));
+	select_node->get_ok_button()->set_text(TTR("Pick"));
+
+	VBoxContainer *main_vb = memnew(VBoxContainer);
+	add_child(main_vb);
+
+	GridContainer *gc = memnew(GridContainer);
+	main_vb->add_child(gc);
+	gc->set_columns(2);
+
+	{
+		Label *label = memnew(Label(TTR("Root Type:")));
+		gc->add_child(label);
+		label->set_v_size_flags(Control::SIZE_SHRINK_BEGIN);
+
+		VBoxContainer *vb = memnew(VBoxContainer);
+		gc->add_child(vb);
+
+		node_type_group.instantiate();
+
+		node_type_2d = memnew(CheckBox);
+		vb->add_child(node_type_2d);
+		node_type_2d->set_text(TTR("2D Scene"));
+		node_type_2d->set_button_group(node_type_group);
+
+		node_type_3d = memnew(CheckBox);
+		vb->add_child(node_type_3d);
+		node_type_3d->set_text(TTR("3D Scene"));
+		node_type_3d->set_button_group(node_type_group);
+
+		node_type_gui = memnew(CheckBox);
+		vb->add_child(node_type_gui);
+		node_type_gui->set_text(TTR("User Interface"));
+		node_type_gui->set_button_group(node_type_group);
+
+		HBoxContainer *hb = memnew(HBoxContainer);
+		vb->add_child(hb);
+
+		node_type_custom = memnew(CheckBox);
+		hb->add_child(node_type_custom);
+		node_type_custom->set_button_group(node_type_group);
+
+		Control *spacing = memnew(Control);
+		hb->add_child(spacing);
+		spacing->set_custom_minimum_size(Size2(4 * EDSCALE, 0));
+
+		root_custom_type = memnew(LineEdit);
+		hb->add_child(root_custom_type);
+		root_custom_type->set_placeholder(TTR("Custom Node"));
```

Comment:
```suggestion
		root_other_type->set_placeholder(TTR("Other Node"));
```
"Custom" here might give the impression that it only allows user-created node types.

Reply:
Funny that the root option in empty scene was also called "Custom Node" until I changed it in #26970